src/TenancyMailManager.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy;
+
+use Illuminate\Mail\MailManager;
+
+class TenancyMailManager extends MailManager
+{
+    public static array $mailersToNotCache = [
+        'smtp',
+    ];
+
+    protected function get($name)
+    {
+        if (in_array($name, static::$mailersToNotCache)) {
+            return $this->resolve($name);
+        }
```

Comment:
Is `$this->resolve` the logic that re-resolves the mailer?

It'd be good to add comments here, it's not very clear how this class works.

Reply:
> Is $this->resolve the logic that re-resolves the mailer?

Yeah.

> It'd be good to add comments here, it's not very clear how this class works.

Right, I'm going to add the comments

Reply:
Also, `$mailersToNotCache` seems inaccurate. The mailers are still cached in the `$mailers` property, but even when they are cached, they get resolved again. So maybe something like `$mailersToAlwaysResolve` would be more accurate

Reply:
https://github.com/archtechx/tenancy/pull/989/commits/64aa0c895b6656953a73066b300cb3974741c68b @stancl could you check out the changes I made in this commit?

Reply:
`$mailersToRefresh` or something sounds reasonable. Or even `$tenantMailers`.

I agreed with your previous suggestion (mailers to not cache) since I haven't seen the code and it sounded logical. But here it'd make more sense to make it something like `$tenantMailers` since that's what makes sense to the developer.

"X to not cache" sounds like it's used for *exceptions* or making things exempt. Whereas `$tenantMailers` / `$refreshedMailers` / ... sound like they're used for the *main thing* that this feature does.

Reply:
Yeah, I like `$tenantMailers`

Reply:
I renamed the variable to `$tenantMailers`. Could you look at the comments in this PR and let me know if you'd change anything?